homeassistant/components/vicare/climate.py

```diff
@@ -230,7 +230,10 @@ def update(self) -> None:
     @property
     def hvac_mode(self) -> HVACMode | None:
         """Return current hvac mode."""
-        return VICARE_TO_HA_HVAC_HEATING.get(self._current_mode)
+        if self._current_mode is None:
+            return None
+        mode: str = self._current_mode
+        return VICARE_TO_HA_HVAC_HEATING.get(mode)
```

Comment:
```suggestion
        return VICARE_TO_HA_HVAC_HEATING.get(mode, None)
```

Reply:
```suggestion
        return VICARE_TO_HA_HVAC_HEATING.get(self._current_mode, None)
```

Reply:
If think the None check has to be there still right?